src/components/network/line-layer.js

```diff
@@ -53,18 +54,51 @@ class LineLayer extends CompositeLayer {
                     .map(e => { return { nominalVoltage: e[0], lines: e[1] };})
                     .sort((a, b) => b.nominalVoltage - a.nominalVoltage);
 
-                // add arrows
+                compositeData.forEach(compositeData => {
+                    let lineMap = new Map();
+                    compositeData.lines.forEach(line => {
+                        const positions = props.geoData.getLinePositions(props.network, line, props.lineFullPath);
+                        const cumulativeDistances = props.geoData.getLineDistances(positions);
+                        lineMap.set(line.id, {positions: positions, cumulativeDistances: cumulativeDistances, line: line});
+                    });
+                    compositeData.lineMap = lineMap;
+                });
 
+                // add arrows
                 compositeData.forEach(compositeData => {
+                    compositeData.activePower = [];
                     // create one arrow each DISTANCE_BETWEEN_ARROWS
+                    const lineMap = compositeData.lineMap;
                     compositeData.arrows = compositeData.lines.flatMap(line => {
                         // calculate distance between 2 substations as a raw estimate of line size
-                        const positions = props.geoData.getLinePositions(props.network, line, false);
-                        const lineDistance = getDistance({latitude: positions[0][1], longitude: positions[0][0]},
-                                                         {latitude: positions[1][1], longitude: positions[1][0]});
-
-                        const arrowCount = Math.ceil(lineDistance / DISTANCE_BETWEEN_ARROWS);
-
+                        const directLinePositions = props.geoData.getLinePositions(props.network, line, false);
```

Comment:
Use the real distance now that it's available ?

Reply:
My dev didn't add access to the real distance, it was obtainable before too. There is no point in recalculating the numbers of arrow in each line every time we change the fullLinePath parameter. This approximation is more than  enough to print a good result.

Reply:
let's do this in another PR